src/control/server/ctl_system.go

```diff
@@ -379,13 +379,23 @@ func (svc *ControlService) start(ctx context.Context, rankList []system.Rank) (s
 		results = append(results, hResults...)
 	}
 
-	// in the case of start, don't manually update member state to "started",
-	// member state will transition to "sarted" during join or bootstrap
+	// member state will transition to "started" during join or bootstrap
 	filteredResults := make(system.MemberResults, 0, len(results))
 	for _, r := range results {
-		if r.Errored {
-			filteredResults = append(filteredResults, r)
+		if !r.Errored || r.State != system.MemberStateReady {
+			continue
+		}
+		if !r.Errored && r.State == system.MemberStateReady {
```

Comment:
This will never be true - if !r.Errored we have already continued above.

Reply:
line 385 should read "if !r.Errored && r.State != system.MemberStateReady {" and will move that fix into https://github.com/daos-stack/daos/pull/2498 . without the fix the members in membership will never be updated with state "Ready", this won't affect any functional behaviour apart from in the occasion where a join doesn't complete and in that case the member will have state "Stopped" rather than "Ready"

Reply:
https://github.com/daos-stack/daos/pull/2498/commits/e05906abd2259bb6e81240fc39342bde9d2ff6ff

Reply:
OK. As long as this won't affect functional behavior, I'm OK with putting it off to the follow-on PR.